RecoEgamma/EgammaElectronAlgos/src/ElectronSeedGenerator.cc

```diff
@@ -37,10 +37,8 @@ namespace {
     if (s1.nHits() != s2.nHits())
       return false;
 
-    unsigned int nHits;
-    TrajectorySeed::range r1 = s1.recHits(), r2 = s2.recHits();
-    TrajectorySeed::const_iterator i1, i2;
-    for (i1 = r1.first, i2 = r2.first, nHits = 0; i1 != r1.second; ++i1, ++i2, ++nHits) {
+    auto const &r1 = s1.recHits(), r2 = s2.recHits();
```

Comment:
```suggestion
    auto const &r1 = s1.recHits(), &r2 = s2.recHits();
```
otherwise `r2` will be a copy of `s2.recHits()`.